### ISSUE TITLE
[Request] Add "diary" to the default accepted books for the bookcase

### ISSUE THREAD

**Issue by iTitus** [Nuchaz/BiblioCraft#303]
A diary is kind of a book, so it should go into the bookcase.
Related to AEnterprise/Eureka#39

If you think that this request is to specific, then I would like to request some kind of API to whitelist books (IMC or an API).

**Comment by AEnterprise** [comment]
i think IMC would be best, it's alot easier to use for this and it means no new compile dependencie

**Comment by Nuchaz** [comment]
I'll definitely add it as a default keyword. An API is something I want to do eventually, more for other reasons, it is just one of those things I haven't got around to yet. I am intrigued by this IMC you mention though. Care to possibly enlighten me on what it is and how it works?

**Comment by iTitus** [comment]
It's the abbreviation for InterModCommunication.
During mod loading you can send a message to a specific mod. This feature comes from FML.
The message consists of a key and a value. The value can be a String or NBT.
The receiving mod has to listen to an IMCEvent, which bundles all sent messages.
IMC is often used for whitelisting/blacklisting items from certain features or for registering API using classes, without actually shipping the API.

Further information can be found on the MinecraftForgeWiki or in the source code of Waila, Buildcraft-Additions...

**Comment by iTitus** [comment]
Some useful links:
http://tutorials.darkhax.net/inter-mod-communication.html

https://github.com/BCA-Team/Buildcraft-Additions/blob/2.2.x/src/main/java/buildcraftAdditions/compat/imc/IMCHandler.java
https://github.com/BCA-Team/Buildcraft-Additions/blob/2.2.x/src/main/java/buildcraftAdditions/compat/imc/IMCSender.java

**Comment by Nuchaz** [comment]
Thanks for the details! That looks like a great way for me to add support for this kind of thing. I could even use an IMC based system to allow mods to add support for the furniture paneler so they could have blocks colored as well. Of course, the rendering of the textures and whatnot would have to be handled by the other mod, but it would be an easy way to handle the whitelist for the furniture paneler.

**Comment by iTitus** [comment]
Happy to help you ;)

**Comment by AEnterprise** [comment]
yeah, imc isn't as powerfull as a regular api in terms of flexibility but for things like this it's perfect
